packages/wrapped-keys/src/lib/lit-actions-client/generate-key.ts

```diff
@@ -0,0 +1,42 @@
+import { AccessControlConditions } from '@lit-protocol/types';
+
+import { postLitActionValidation } from './utils';
+import { GeneratePrivateKeyParams } from '../types';
+
+interface GeneratePrivateKeyLitActionParams extends GeneratePrivateKeyParams {
+  pkpAddress: string;
+  accessControlConditions: AccessControlConditions;
+  litActionIpfsCid: string;
+}
+
+interface GeneratePrivateKeyLitActionResult {
+  ciphertext: string;
+  dataToEncryptHash: string;
+  publicKey: string;
+}
+
+export async function generateKeyWithLitAction({
+  litNodeClient,
+  pkpSessionSigs,
+  litActionIpfsCid,
+  accessControlConditions,
+  pkpAddress,
+}: GeneratePrivateKeyLitActionParams): Promise<GeneratePrivateKeyLitActionResult> {
+  try {
+    const result = await litNodeClient.executeJs({
+      sessionSigs: pkpSessionSigs,
+      ipfsId: litActionIpfsCid,
+      jsParams: {
+        pkpAddress,
+        accessControlConditions,
+      },
+    });
+
+    const response = postLitActionValidation(result);
```

Comment:
Does `postLitActionValidation()` implementation looks good?

Reply:
:+1: Yup, I think so - the errors I get from failures seem to be about as good as we can expect given how LIT actions throw.